### PR TITLE
[BUG] - Pin-code current-input id is not set correctly (Issue/2891)

### PR DESCRIPTION
**Pull request by henrijoss**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: https://designsystem.porsche.com/issue/2891

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves [#2891](https://github.com/porsche-design-system/porsche-design-system/issues/2891)

### CODE REVIEW

File: packages/components/src/components/pin-code/pin-code-utils.spec.ts [hunk 1]
```diff
@@ -316,3 +317,40 @@ describe('syncHiddenInput()', () => {
...
+    [2, '123 56', 6, false],
+    [3, '123 56', 6, true],
+    [4, '123 56', 6, false],
+    [5, '123 56', 6, false],
+    [0, '1234', 4, false], // All inputs have some value: set current-input id on the last input element
+    [1, '1234', 4, false],
+    [2, '1234', 4, false],
+    [3, '1234', 4, true],
+  ])(
+    'should return correct current input for index: %d, value: %s and length: %d',
```
Comment (tatjana-gartner2) [inline packages/components/src/components/pin-code/pin-code-utils.spec.ts:350]:
```suggestion
    'should for index: %d, value: %s and length: %d' return: %s,
```

File: packages/components/src/components/pin-code/pin-code-utils.ts [hunk 2]
```diff
@@ -110,3 +110,22 @@ export const isFormSubmittable = (host: HTMLElement, form: HTMLFormElement): boo
...
+ * @param {number} index - The index of the current input element.
+ * @param {string} value - The input value.
+ * @param {number} length - The total number of input elements.
+ * @returns {boolean} - True if the current input should have a specific identifier, indicating it corresponds to the first non-filled input; otherwise, false.
+ */
+export const isCurrentInput = (index: number, value: string, length: number): boolean => {
+  if (!value) {
+    return index === 0; // No value entered at all: set current-input id on the first input element
+  }
+  const firstWhitespaceIndex = value.split('').findIndex((char) => char === ' ');
```
Comment (denyo) [inline packages/components/src/components/pin-code/pin-code-utils.ts:126]: I think `value.indexOf(' ')` would be easier and shorter.

File: packages/components/CHANGELOG.md [hunk 3]
```diff
@@ -67,6 +67,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 #### Fixed
 
+- `Pin Code`: `current-input` id is not set correctly
```
Comment (denyo) [inline packages/components/CHANGELOG.md:70]: From a consumer perspective I would have no clue what this means. Instead describing the focus behavior that was improved would be better.

**Comment by denyo** [comment]
Should we add e2e tests for the different scenarios that are described in the ticket?

**Comment by henrijoss** [comment]
> Should we add e2e tests for the different scenarios that are described in the ticket?

There already is an e2e test for checking if the label click works when the `current-input` id is set. Testing if the correct input has the id should be covered with the unit test.

**Review by denyo — APPROVED** [review]

**Pull request merged**